deploy/runtools/firesim_topology_with_passes.py

```diff
@@ -26,10 +27,22 @@
 
 @parallel
 def instance_liveness() -> None:
-    """ Confirm that all instances are accessible (are running and can be ssh'ed into) first so that we don't run any
-    actual firesim-related commands on only some of the run farm machines."""
+    """ Confirm that all instances are accessible (are running and can be
+    ssh'ed into) first so that we don't run any actual firesim-related commands
+    on only some of the run farm machines.
+
+    Also confirm that the default shell in use is one that is known to handle
+    commands we pass to run in the manager.
+    """
     rootLogger.info("""[{}] Checking if host instance is up...""".format(env.host_string))
     run("uname -a")
+    collect = run("echo $SHELL")
+
+    allowed_shells = ["bash"]
```

Comment:
Do we not have control over this based on how we have configured fabric?  I'm surprised that we depend on the login shell.

We should explicitly specify which shell(s) are supported and configure fabric to use the supported shell explicitly rather than requiring the user to set their login shell to be something that works with the commands the manager is passing to fabric.

Reply:
As far as I can tell, it is not possible to change which shell ssh uses apart from changing the default shell: https://serverfault.com/questions/162018/force-ssh-to-use-a-specific-shell

Reply:
I've added a long comment to explain the problem and changed the implementation to:

do nothing for good shells
exit(1) and print error for known bad shells
print a warning for unknown shells

I have not had anyone tell me they've successfully used zsh, so it's in the "unknown" category for now.